File: massenger/app/src/main/java/com/example/massenger/SettingsViewModel.java
package com.example.massenger;

import android.util.Log;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class SettingsViewModel extends ViewModel {

    private static final String TAG = "SettingsViewModel";

    private MutableLiveData<Boolean> isDarkTheme = new MutableLiveData<>(false);
    private MutableLiveData<Boolean> notificationsEnabled = new MutableLiveData<>(true);

    public SettingsViewModel() {
        Log.d(TAG, "SettingsViewModel создан");
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        Log.d(TAG, "SettingsViewModel уничтожен");
    }
    public LiveData<Boolean> getIsDarkTheme() {
        return isDarkTheme;
    }
    public LiveData<Boolean> getNotificationsEnabled() {
        return notificationsEnabled;
    }
    public void setIsDarkTheme(boolean darkTheme) {
        Log.d(TAG, "Тема изменена на: " + (darkTheme ? "темная" : "светлая"));
        isDarkTheme.setValue(darkTheme);
    }
    public void setNotificationsEnabled(boolean enabled) {
        Log.d(TAG, "Уведомления: " + (enabled ? "включены" : "выключены"));
        notificationsEnabled.setValue(enabled);
    }
    public void toggleTheme() {
        Boolean current = isDarkTheme.getValue();
        if (current != null) {
            setIsDarkTheme(!current);
        }
    }
}

File: app/src/main/java/com/example/messengerlab1/SettingsViewModel.kt
package com.example.messengerlab1

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class SettingsViewModel : ViewModel() {

    private val _darkTheme = MutableLiveData(false)
    val darkTheme: LiveData<Boolean> = _darkTheme

    init {
        Log.d("VM", "SettingsViewModel init")
    }

    fun setDarkTheme(enabled: Boolean) {
        _darkTheme.value = enabled
        Log.d("VM", "SettingsViewModel darkTheme=$enabled")
    }

    override fun onCleared() {
        super.onCleared()
        Log.d("VM", "SettingsViewModel onCleared")
    }
}